server/channels/store/sqlstore/channel_store.go

```diff
@@ -2021,6 +2021,92 @@ func (s SqlChannelStore) GetChannelMembersTimezones(channelId string) ([]model.S
 	return dbMembersTimezone, nil
 }
 
+func (s SqlChannelStore) GetChannelsWithUnreadsAndWithMentions(ctx context.Context, channelIDs []string, userID string) ([]string, []string, map[string]int64, error) {
+	query := s.getQueryBuilder().Select(
+		"Channels.Id",
+		"Channels.Type",
+		"Channels.TotalMsgCount",
+		"Channels.LastPostAt",
+		"ChannelMembers.MsgCount",
+		"ChannelMembers.MentionCount",
+		"ChannelMembers.NotifyProps",
+		"ChannelMembers.LastViewedAt",
+		"Users.NotifyProps",
+	).
+		From("ChannelMembers").
+		InnerJoin("Channels ON ChannelMembers.ChannelId = Channels.Id").
+		InnerJoin("Users ON ChannelMembers.UserId = Users.Id").
+		Where(sq.Eq{
+			"ChannelMembers.ChannelId": channelIDs,
+			"ChannelMembers.UserId":    userID,
+		})
+
+	queryString, args, err := query.ToSql()
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "channel_tosql")
+	}
+
+	rows, err := s.GetReplicaX().DB.Query(queryString, args...)
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "failed to find ChannelMembers, TeamScheme and ChannelScheme data")
+	}
+	defer deferClose(rows, &err)
+
+	channelsWithUnreads := []string{}
+	channelsWithMentions := []string{}
+	readTimes := map[string]int64{}
+
+	for rows.Next() {
+		var id string
+		var channelType string
+		var totalMsgCount int
+		var lastPostAt int64
+		var msgCount int
+		var mentionCount int
+		var channelNotifyProps model.StringMap
+		var lastViewedAt int64
+		var userNotifyProps model.StringMap
+
+		err = rows.Scan(&id, &channelType, &totalMsgCount, &lastPostAt, &msgCount, &mentionCount, &channelNotifyProps, &lastViewedAt, &userNotifyProps)
```

Comment:
Not sure if this is the best way to read from the results, but open to any suggestions.

Reply:
Is there any specific reason we're iterating over the rows? You can use something like
```go
var yourData []dataType
s.GetReplicaX().Select(&yourData, queryString, args...)
```

There are some examples in the file. Take a look at, for example, `GetPinnedPosts`

Reply:
Not a problem from my side in creating an ad-hoc type, I think that improves the readability of the code.